### PR TITLE
latex2html: fix executable path

### LINKED ISSUE #25024: latex2html configuration is missing
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

# Problems

After installing latex2html by `brew install latex2html`.  The configuration file `/usr/local/Cellar/latex2html/2018/l2hconf.pm` does not contain required pathname:

```
$LATEX = '';   # LaTeX
$DVIPS = '';   # dvips
$DVIPNG = ''; # dvipng
$TEX = "";                       # TeX
$INILATEX = " \"&latex\"";    # initex+latex
```

Note that LaTeX (MacTeX) was already installed on my system, LaTeX binaries are available in the `PATH`:

```
$ which latex
/Library/TeX/texbin/latex
```

1.  It would be nice that homebrew actually update `l2hconf.pm` accordingly.  For example:

```
$LATEX = '/Library/TeX/texbin/latex';   # LaTeX
$DVIPS = '/Library/TeX/texbin/dvips';   # dvips
$DVIPNG = '/Library/TeX/texbin/dvipng'; # dvipng
$TEX = "/Library/TeX/texbin/TeX";                       # TeX
$INILATEX = "/Library/TeX/texbin/initex \"&latex\"";    # initex+latex
```

2. or, print what to do on `brew info latex2html` something like:

```
Note that you must update some of the variable in 
`/usr/local/Cellar/latex2html/2018/l2hconf.pm` at least for
$LATEX, $DVIPS, $DVIPNG, $TEX and $INILATEX.
```

# outputs

## output of brew gist-logs latex2html

        Error: No logs.

## output of brew config

```
HOMEBREW_VERSION: 1.5.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: 08f7e8c5887d1c654a1a5d5bc851a9c9d5a00371
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c340723a0a851423a451ad1fd959636afdf2ec10
Core tap last commit: 23 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_RUBY_WARNINGS: -W0
CPU: octa-core 64-bit kabylake
Homebrew Ruby: 2.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Clang: 9.0 build 900
Git: 2.14.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/bin/python3.6
Ruby: /Users/seong-kookshin/.rbenv/shims/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Java: 1.8.0_144
macOS: 10.13.3-x86_64
Xcode: 9.2
CLT: 9.2.0.0.1.1510905681
X11: 2.7.11 => /opt/X11
```

## output of brew doctor

There are some warnings, but none of them are relevant for this issue:

```
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Putting non-prefixed coreutils in your path can cause gmp builds to fail.

Warning: Homebrew's sbin was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting the PATH for example like so
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.bash_profile

Warning: Broken symlinks were found. Remove them with `brew prune`:
  /usr/local/bin/gambcomp-C.bat
  /usr/local/bin/gambdoc.bat
```

### PR DESCRIPTION
**Pull request by holymonson**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
fix https://github.com/Homebrew/homebrew-core/issues/25024

### CODE REVIEW

File: Formula/latex2html.rb [hunk 1]
```diff
@@ -4,23 +4,26 @@ class Latex2html < Formula
...
   sha256 "095b6d43599506aa0936b548ee92c7742c8b127d64e3829000f7681a254a7917"
 
-  bottle do
-    cellar :any_skip_relocation
-    sha256 "2a8631d720a4628e09e09448e866504d01cb3a26e8ea7bc20620199e01b0cbed" => :mojave
-    sha256 "488549a6634ef38ec4c7493e1649d2754d40fa1b200fd9f2ab33ec12e1fd6572" => :high_sierra
-    sha256 "d3019ab7c34e06538dd3c24c4fae92bb212abf165144e84589bf8af9af54fdc3" => :sierra
-  end
+  # must build from source to configure executable path.
+  bottle :unneeded
```
Comment (fxcoudert) [inline Formula/latex2html.rb]: No, this should not hardcode a path.

File: Formula/latex2html.rb [hunk 2]
```diff
@@ -4,23 +4,26 @@ class Latex2html < Formula
...
 
   def install
+    ENV.prepend_path "PATH", "/Library/TeX/texbin"
     system "./configure", "--prefix=#{prefix}",
-                          "--without-mktexlsr",
                           "--with-texpath=#{share}/texmf/tex/latex/html"
     system "make", "install"
   end
 
+  def caveats; <<~EOS
```
Comment (fxcoudert) [inline Formula/latex2html.rb]: Again, no. This is not acceptable.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by holymonson** [comment]
latex2html depands on a TeX system (TeXLive), on macOS, it's MacTex (the only one). But it's in homebrew-cask instead of homebrew-core, we can't draw the dependence in formula. And this package is pure code, no need to bottle at the very start.

Unless include mactex into brew-core, there is no way to elegantly fix it. Now latex2html is broken all the way, if it's not acceptable, please remove this formula rather then leaving a trap.

**Comment by fxcoudert** [comment]
> Now latex2html is broken all the way

We welcome a fix that follows the rules of homebrew formulas.

> this package is pure code, no need to bottle at the very start

There is a `configure && make` process, and the files produced are system-dependent (which we know because the bottles on different macOS versions have different checksums). So it should be bottled.

> latex2html depands on a TeX system (TeXLive), on macOS, it's MacTex

Miktex can be installed on macOS. TeXLive can be installed in non-default paths. We need to consider these possibilities.

**Comment by fxcoudert** [comment]
Confirmed by installing locally that software works fine by setting `l2hconf.pm` manually. That `latex2html` cannot work by autodetecting installed `latex` binaries when they are in `PATH` is their bug to fix, it's actually been reported but the project appears inactive: https://github.com/latex2html/latex2html/issues/22

**Pull request closed without merging**